pkg/query-service/app/http_handler.go

```diff
@@ -542,7 +559,7 @@ func (aH *APIHandler) addTemporality(ctx context.Context, qp *v3.QueryRangeParam
 	if qp.CompositeQuery != nil && len(qp.CompositeQuery.BuilderQueries) > 0 {
 		for name := range qp.CompositeQuery.BuilderQueries {
 			query := qp.CompositeQuery.BuilderQueries[name]
-			if query.DataSource == v3.DataSourceMetrics {
+			if query.DataSource == v3.DataSourceMetrics && query.Temporality == "" {
```

Comment:
you are trying to fill only empty temporality values in the query payload? Why would frontend send empty Temporality for only a few of the metric names for which you need to query DB to populate it.

Reply:
had a chat in slack